src/libexpr/eval.cc

```diff
@@ -95,11 +95,16 @@ RootValue allocRootValue(Value * v)
 #endif
 }
 
-void Value::print(const SymbolTable & symbols, std::ostream & str,
-    std::set<const void *> * seen) const
+void Value::print(const SymbolTable &symbols, std::ostream &str,
+                  std::set<const void *> *seen, int depth) const
+
 {
     checkInterrupt();
 
+    if (depth <= 0) {
+        str << "«depth»";
```

Comment:
```suggestion
        str << "«too deep»";
```

Not to suggest that there's a representation of some sort of depth concept at that location.